velox/functions/prestosql/Arithmetic.h

```diff
@@ -198,6 +198,30 @@ struct Log10Function {
   }
 };
 
+template <typename T>
+struct Log2FunctionNaNAsNull {
+  FOLLY_ALWAYS_INLINE bool call(double& result, double a) {
+    double yAsymptote = 0.0;
+    if (a <= yAsymptote) {
```

Comment:
Use std::isnan for NaN?

Reply:
Did you mean using `std::isnan(a)` to replace `a <= yAsymptote` ? The two don't seem to share the same semantic?

Reply:
btw NaN is the output of log(-1) for example.

Reply:
Got it. With above logic, will `null` be returned for this case instead of `NaN`?

Reply:
Yes, Velox treats the return type to be nullable if `bool call(...)` is used here.